build.zig

```diff
@@ -1688,29 +1688,21 @@ fn strip_root_module(root_module: *std.Build.Module) void {
 fn set_windows_dll(allocator: std.mem.Allocator, java_home: []const u8) void {
     comptime assert(builtin.os.tag == .windows);
 
-    // Declaring the function with an alternative name because `CamelCase` functions are
-    // by convention, used for building generic types.
-    const set_dll_directory = @extern(
-        *const fn (path: [*:0]const u8) callconv(.C) std.os.windows.BOOL,
-        .{
-            .library_name = "kernel32",
-            .name = "SetDllDirectoryA",
-        },
-    );
-
     const java_bin_path = std.fs.path.joinZ(
         allocator,
         &.{ java_home, "\\bin" },
     ) catch unreachable;
-    _ = set_dll_directory(java_bin_path);
+    _ = SetDllDirectoryA(java_bin_path);
 
     const java_bin_server_path = std.fs.path.joinZ(
         allocator,
         &.{ java_home, "\\bin\\server" },
     ) catch unreachable;
-    _ = set_dll_directory(java_bin_server_path);
+    _ = SetDllDirectoryA(java_bin_server_path);
 }
 
+extern "kernel32" fn SetDllDirectoryA(path: [*:0]const u8) callconv(.C) std.os.windows.BOOL;
```

Comment:
Why not declare _all_ windows API we need in `stdx.windows`? (here and jni_thread_cleaner.zig`)
We can even simplify the `tidy.zig` validation to just skip this entire file instead of looking for `extern "kernel32"`.

Reply:
> This is build.zig, and, while we technically can @import("../src/stdx.zig") here, I'd rather avoid depending on stdx in our build.zig, to avoid circularities.

:sweat_smile: I just used `stdx.cut` in build.zig in another PR

> ... if you think it's a good idea! I don't have a good feel for our client library dependencies.

Well, clients include `stdx` (from `vrs.stdx`). If this  API were already declared in `stdx.windows`, I'd use it instead of redeclaring it to avoid referencing `stdx` ... but I don't have a strong opinion here, especially because we want to make `stdx` an independent module.